### PR TITLE
Simplify menu API to add tools at runtime

### LINKED ISSUE #141: simpler menu API to add new tools to toolbar, even at runtime
As more tools are added, we might use a simpler menu API to add new tools, even after initialization, like, perhaps:

```js
img = new L.DistortableImageOverlay(...);
img.addAction(
  EditOverlayAction.extend({ 
 	initialize: function(map, overlay, options) { 
 		var icon = 'rotate-left'; 
 		options = options || {}; 
 		options.toolbarIcon = { 
 			html: '<span class="fa fa-' + icon + '"></span>', 
 			tooltip: 'Rotate', 
 			title: 'Rotate'	 
 		}; 
 		EditOverlayAction.prototype.initialize.call(this, map, overlay, options); 
 	}, 
 	addHooks: function() { 
 		var editing = this._overlay.editing; 
 		editing._toggleRotateDistort(); 
 		this.disable(); 
 	} 
 })
);
```

The toolbar is built using https://github.com/Leaflet/Leaflet.toolbar, so we can look into how that library works. The API reference is here: https://github.com/leaflet/Leaflet.Toolbar/wiki/API-Reference#ltoolbar2action

### PR DESCRIPTION
**Pull request by rexagod**
Fixes #141.

peek 2019-02-23 06-46

### CODE REVIEW

**Comment by rexagod** [comment]
#126

**Comment by rexagod** [comment]
`_removeTool` was added in #155

File: src/DistortableImageOverlay.js [hunk 1]
```diff
@@ -69,6 +70,15 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
 		});
 	},
 
+	_addTool: function(tool) {
+		this._toolArray.push(tool);
+		L.DistortableImage.EditToolbar = LeafletToolbar.Popup.extend({
```
Comment (jywarren) [inline src/DistortableImageOverlay.js]: So, this replaces the existing toolbar with a new one from the toolArray? Cool.

File: src/edit/DistortableImage.Edit.js [hunk 2]
```diff
@@ -1,361 +1,390 @@
...
-			69: '_toggleIsolate', // e
-			73: '_toggleIsolate', // i
-			76: '_toggleLock', // l
-			79: '_toggleOutline', // o
-			82: '_toggleRotateDistort', // r
-      84: '_toggleTransparency', // t
+  options: {
+    opacity: 0.7,
+    outline: "1px solid red",
+    keymap: {
```
Comment (jywarren) [inline src/edit/DistortableImage.Edit.js]: This cleanup is super great!

File: src/edit/DistortableImage.Edit.js [hunk 3]
```diff
@@ -1,361 +1,390 @@
...
-		else { point = target._leaflet_pos; }
-		var raised_point = map.containerPointToLatLng(new L.Point(point.x,point.y-20));
-		raised_point.lng = overlay.getCenter().lng;
-		if (this._overlay.options.suppressToolbar !== true) {
-		this.toolbar = new L.DistortableImage.EditToolbar(raised_point).addTo(map, overlay);
-		overlay.fire('toolbar:created');
-		}
-
-		L.DomEvent.stopPropagation(event);
+  // _demonstrator: function() {
```
Comment (jywarren) [inline src/edit/DistortableImage.Edit.js]: Shall we tidy this up or add comments to explain? Thanks!

File: src/edit/DistortableImage.EditToolbar.js [hunk 4]
```diff
@@ -1,125 +1,135 @@
...
+var defaults = [
+  ToggleTransparency,
+  RemoveOverlay,
+  ToggleOutline,
+  ToggleEditable,
+  ToggleRotateDistort,
+  ToggleExport
+];
+
+window.defaults = defaults;
```
Comment (jywarren) [inline src/edit/DistortableImage.EditToolbar.js]: What if we assigned this to `L.DistortableImage.EditToolbarDefaults` instead of the window?

**Comment by jywarren** [comment]
@sashadev-sky check this out, it's really nice! It also gives us a hint to the implementation of different UIs, in terms of how we structure that.

File: src/edit/DistortableImage.EditToolbar.js (hunk 4 above)
Comment (rexagod) [reply src/edit/DistortableImage.EditToolbar.js]: Initially I did the same, but forgot to revert back to it after some testing on `window`. Sorry! I've made the required changes. :smile:

**Comment by jywarren** [comment]
OK, looks great and I believe ready to merge once rebased!

**Comment by rexagod** [comment]
@jywarren I've rebased this branch. Please review. Thanks!

**Comment by jywarren** [comment]
Great! Super work here. Thanks!

**Pull request merged**